### DISCUSSION TITLE
Huge coverage gap between jest and vitest

### DISCUSSION THREAD

**Discussion by tqn-treezor** [vitest-dev/vitest#1240] [General]
Hello everyone! Before I open an issue, I wanted to make sure I wasn't doing anything wrong: I just made the transition from Jest to Vitest, and observed some huge coverage gaps.
In order to try and reduce these, I tried to align my former `jest.config.js` with my new `vitest.config.ts`.

This is my `jest.config.js`:
```js
const jestConfig = {
  preset: 'ts-jest',
  globals: {
    'ts-jest': {
      useESM: true,
    },
  },
  extensionsToTreatAsEsm: ['.ts'],
  transform: {
    '^.+\\.ts?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  moduleNameMapper: {
    '@treezor/sdk/(.*)': '<rootDir>/$1',
  },
  testPathIgnorePatterns: ['dist', 'coverage'],
  testEnvironment: 'node',
  collectCoverage: true,
  collectCoverageFrom: [
    './src/**/*.{js,jsx,ts,tsx}',
    '!./src/**/index.{js,ts}',
    '!./src/mock/*.{js,ts}',
  ],
  coverageThreshold: {
    global: {
      statements: 90.7,
      branches: 38.04,
      lines: 90.53,
      functions: 83.65,
    },
  },
};

export default jestConfig;
```

And this is my `vite.config.ts`:
```typescript
import { defineConfig } from 'vitest/config';

export default defineConfig({
  test: {
    coverage: {
      reporter: ['text', 'json', 'html'],
      extension: ['js', 'jsx', 'ts', 'tsx'],
      include: ['**/src/**/*.{js,jsx,ts,tsx}'],
      exclude: [
        '**/node_modules/**',
        '**/dist/**',
        '**/cypress/**',
        '**/.{idea,git,cache,output,temp}/**',
        '**/coverage/**',
        './src/**/index.{js,ts}',
        './src/mock/*.{js,ts}',
        '**/approvals/**',
      ],
    },
    exclude: [
      '**/node_modules/**',
      '**/dist/**',
      '**/cypress/**',
      '**/.{idea,git,cache,output,temp}/**',
      '**/coverage/**',
    ],
  },
});
```

As you can see, my coverages using jest were the following:
```
statements: 90.7,
branches: 38.04,
lines: 90.53,
functions: 83.65,
```

When running vitest, I get the following coverages:
```
statements: 97.25,
branches: 95.96,
lines: 97.25,
functions: 88.53,
```

The biggest gap observed is with the branches: I jump from 38.04% to 95.96%. Has anyone observed a similar gap?

**Comment by tqn-treezor** [comment]
(If you see anything unusual that may explain such gap differences, I'm also totally open to suggestions ^^)

**Reply by sebastianrothe** [reply]
It might be the `--all` flag. It is `false` by default and this means, only referenced files are counted for the coverage results. You also need to supply `--src` with a list of folders to be scanned.

https://github.com/bcoe/c8#cli-options--configuration

**Comment by tqn-treezor** [comment]
I am going to leave that discussion, but for those who come here after me, here are a few things I found out:
- types.ts were ignored by default in Jest and not in Vitest
- *.entity.ts were ignored by default in Jest and not in Vitest
- *.test.ts were ignored by default in Jest and not in Vitest
- For includes/excludes, Vitest takes absolute paths, Jest accepts relative paths (that may lead to include or exclude paths not working and giving erroneous results in your coverage)
- Comments are ignored by default in Jest and not in Vitest
- Import lines are ignored by default in Jest and not in Vitest

Also, all that coverage part is managed by C8 and not Vitest itself - so if you have any issue to report, or any feature request, they should be asked there instead: https://github.com/bcoe/c8
